src/react/accordion/Accordion.tsx

```diff
@@ -24,52 +29,81 @@ export interface AccordionProps {
   showSectionText?: string
 }
 
-export const Accordion = forwardRef<HTMLDivElement, AccordionProps & JSX.IntrinsicElements['div']>(({
-  children,
-  className,
-  headingTag = 'h2',
-  hideAllSectionsText,
-  hideSectionText,
-  hideSectionAriaLabelText,
-  id,
-  rememberExpanded,
-  showAllSectionsText,
-  showSectionText,
-  showSectionAriaLabelText,
-  ...rest
-}, forwardedRef) => {
-  const { ref } = useMODUKComponent('Accordion')
+function getRemountProps(props: AccordionProps) {
+  const rootProps = pick(props, [
+    'headingTag',
+    'hideAllSectionsText',
+    'hideSectionAriaLabelText',
+    'hideSectionText',
+    'rememberExpanded',
+    'showAllSectionsText',
+    'showSectionAriaLabelText',
+    'showSectionText',
+  ])
+  const childrenProps = flattenChildren(props.children).filter(isValidElement<AccordionItemProps>).map((
+    child,
+  ) => pick(child.props, ['heading', 'summary', 'expanded']))
+
+  return [rootProps, childrenProps]
+}
+
+export const Accordion = forwardRef<HTMLDivElement, AccordionProps & JSX.IntrinsicElements['div']>(
+  (props, forwardedRef) => {
+    const { ref } = useMODUKComponent('Accordion')
+    const key = useRef<number>(0)
+    const remountProps = getRemountProps(props)
+    const previousRemountProps = usePrevious(remountProps)
+
+    if (previousRemountProps !== remountProps) {
+      key.current += 1
```

Comment:
Would this ever be equal? Pick create a new object so I think we would have to use something like https://www.npmjs.com/package/shallow-equal.

Reply:
Thanks, I was using lodash `isEqual` but accidentally removed it :disappointed: :facepalm: 

Added it back in and added a test for that scenario.